pkgs/applications/editors/neovim/wrapper.nix

```diff
@@ -30,13 +30,23 @@ let
     # set to false if you want to control where to save the generated config
     # (e.g., in ~/.config/init.vim or project/.nvimrc)
     , wrapRc ? true
-    , neovimRcContent ? ""
+    , neovimRcContent ? null
+    , neovimLuaRcContent ? null
     # entry to load in packpath
     , packpathDirs
     , ...
   }:
+
+  # we can not have both, a lua and a vimL init file, at least one needs to be
+  # null
+  assert isNull neovimRcContent || isNull neovimLuaRcContent;
+
   let
 
+    selectLua = isNull neovimRcContent;
+    rcContent = if !selectLua then "source ${writeText "init.vim" neovimRcContent}"
```

Comment:
rcContent should always be neovimLuaRcContent AND `source "init.vim" with neovimRcContent` (order might matter, I would put lua first because I expect it to become the default) this way we dont break existing setups by ignoring the viml content